docs/source/developer_guide/testing_framework.rst

```diff
@@ -0,0 +1,269 @@
+low level tests that check correctness of the estimator.
+These typically use only the simplest idioms in ``pytest`` (e.g., fixture parameterization).
+Adding a new estimator will typically *not* require changes to module level tests,
+as new estimators are automatically discovered and looped over by the existing tests.
+* Introducing or changing base class level interface points will typically require addition of module level tests,
+and addition of, or modification to scenarios with functionality specific to these interface points.
+Rarely, this may require changes package level tests.
+* Major interface changes or addition of modules may require writing of entire test suites,
+and changes or additions to package level tests.
+
+Adding low level tests
+~~~~~~~~~~~~~~~~~~~~~~
+
+Low level tests are "free-form" and should follow best ``pytest`` practice.
+``pytest`` tests should be located in the appropriate ``tests`` folder of the module where a change is made.
+Examples should be located in the docstring of the class or function added.
+
+For an added estimator of name ``estimator_name``, the test file should be called ``test_estimator_name.py``.
+
+Useful functionality to write tests:
+* example fixture generation, via ``datatypes.get_examples``
+* data format checkers in ``datatypes``: ``check_is_mtype``, ``check_is_scitype``, ``check_raise``
+* miscellaneous utilities in ``utils``, especially in ``_testing``
+
+Escaping tests
+~~~~~~~~~~~~~~
+
+On occasion, it may make sense to escape individual estimators from individual tests.
+
+This can be done (currently, as of 0.9.0) in two ways:
+
+* adding the estimator or test/estimator combination to the ``EXCLUDED_TESTS`` or ``EXCLUDE_ESTIMATORS`` in the appropriate ``_config`` file.
+* adding a check condition in the ``is_excluded`` method used in ``pytest_generate_fixtures``, possibly only if the testing module supports this
+
+Escaping tests directly in the tests, e.g., via ``if isinstance(estimator_instance, MyClass)`` should be avoided where possible.
+
+Adding package or module level tests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Module level tests use ``pytest_generate_tests`` to define fixtures.
+
+The available fixtures vary per module, and are listed in the docstring of ``pytest_generate_tests``.
+
+A new test should use these fixtures, if possible, but also can add new fixtures via ``pytest`` basic fixture functionality.
+
+If new fixture variables are to be used throughout the module, or depend on existing fixtures,
+instructions in the next section should be followed.
+
+Where possible, scenarios should be used to simulate generic method calls (see above),
+instead of creating and passing arguments directly. Scenarios will ensure consistent coverage of input argument cases.
+
+Adding fixture variables
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+One-off fixture variables can be added using ``pytest`` basic functionality.
```

Comment:
Not much info in this sentence, consider removing

Reply:
Why? This is "option 1", the rest is "option 2". Removing "option 1" removes crucial information imo?

Reply:
I don't find any useful information in this sentence so my suggestion is to remove, but only a suggestion. Feel free to resolve if you think otherwise

Reply:
made it clearer that these are two cases. Better?